test/generators/go/presets/CommonPreset.spec.ts

```diff
@@ -0,0 +1,15 @@
+import {
+  GoGenerator,
+  GO_COMMON_PRESET,
+  GoCommonPresetOptions
+} from '../../../../src/generators';
+
+describe('Go_COMMON_PRESET', () => {
+  let generator: GoGenerator;
+  beforeEach(() => {
+    generator = new GoGenerator({
+      presets: [{ preset: GO_COMMON_PRESET, options: { addJsonTag: true } }]
```

Comment:
@jonaslagoni I am having some issues here with `GO_COMMON_PRESET` 
<img width="1145" alt="Screenshot 2024-03-01 at 7 18 42 PM" src="https://github.com/asyncapi/modelina/assets/41781438/8025d437-cfd9-40a8-931c-b30968884336">

Reply:
Go into /src/generators/go/GoPreset.ts and change the following lines:
```
export type GoPreset<O = any> = Preset<{
  struct: StructPresetType<O>;
  enum: EnumPresetType<O>;
}>;

export const GO_DEFAULT_PRESET: GoPreset<GoOptions> = {
  struct: GO_DEFAULT_STRUCT_PRESET,
  enum: GO_DEFAULT_ENUM_PRESET
};
```

Its a bug with the preset types 🙂